### DISCUSSION TITLE
Stuck on themes - how to mimic GTK shade/mix operators using translucent SVGs?

### DISCUSSION THREAD

**Discussion by AzumaHazuki** [lxqt/lxqt#2508] [Q&A]
Yes, this _is_ germane to LXQt. I've been trying to make themes for us for Mint-Y, Shiki-Colors, and Murrine-Colors, and am not able to replicate the color gradients produced on prelight/urgent/etc buttons.

Inside the .gtkrc and gtk.css files, there are directives like `gradient_shades(1.09,1.01,1.01,0.91)` and `bg[PRELIGHT] = shade(0.85, @selected_bg_color)`. 

In LXQt terms, this says to me I should be doing something like

```
#TaskBar QToolButton:hover {
color: palette(highlighted_text);
background: palette(highlight);
background-origin: content;

border: 3px solid transparent;
border-image: url(assets/buttons/hover.svg);
}
```
where `hover.svg` is an SVG file I made with a 4-stop gradient: 0 (255,255,255,9%), 49 (255,255,255,1%), 50 (255,255,255,1%), 100 (0,0,0,9%) and a lower layer of (0,0,0,15%). On the panel, this says "if you're hovering over a taskbar button, paint it the user's highlight color, then overlay this .svg which will darken it by 15% and then, overtop of that, apply that glassy gradient."

Problem is, this doesn't work. The shape is correct, but the colors are wrong, and it also doesn't produce that nice sharp midline in the gradient the Murrine themes are known for.

Does anyone here know what to do? Is the GTK `shade` function using something other than the rgba colorspace to generate its gradients? This is really frustrating.

**Comment by tsujan** [comment]
> Yes, this is germane to LXQt.

Not an answer to your question, but to prevent a probable misunderstanding:

LXQt doesn't invent styling; it just uses Qt's [stylesheets](https://doc.qt.io/qt-6/stylesheet-reference.html) for its Panel and Runner.

**Reply by AzumaHazuki** [reply]
Uh...yes, I'm aware. What does that have to do with the OP though?

**Reply by tsujan** [reply]
> What does that have to do with the OP though?

It was a side note, telling that this isn't specific to LXQt.

As for the answer, you might find it in the linked doc: https://doc.qt.io/qt-6/stylesheet-reference.html#border-image

**Reply by AzumaHazuki** [reply]
Yes, QSS supports HSV and HSL colorspaces. But you can't specify more than one `background-color` property, correct? I'd need `palette(highlight);` _and_ `qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 hsla(0, 255, 100%, 9%), stop:0.49 hsla(0, 255, 100%, 1%), stop:0.5 hsla(0, 255, 100%, 1%), stop:1 hsla(0, 0, 100%, 9%);` for example.

**Reply by tsujan** [reply]
You said, "The shape is correct, but the colors are wrong...". The link may explain why: "A border image is an image that is composed of nine parts....".

**Reply by AzumaHazuki** [reply]
And it turns out that if you DON'T the trailing number at the end of a directive like `border-image = url(path/to/some.svg) 3;` it applies the middle of the image as an overlay.

I'm not a programmer, but I'm also not dumb, okay? Girls can computer too. I'm trying to figure out what gradient I would need to make in Inkscape or the GIMP to produce Murrine-like shading with a sharp stop in the middle, and how to do it. Lightness isn't Value isn't Luma isn't Luminance, and GTK internally uses HSL.
